march_rqt_robot_monitor/src/march_rqt_robot_monitor/diagnostic_analyzers/temperature.py

```diff
@@ -0,0 +1,85 @@
+
+from diagnostic_msgs.msg import DiagnosticStatus
+import rospy
+from sensor_msgs.msg import Temperature
+
+
+LOWER_THRESHOLD_VALID_VALUE = 0
+UPPER_THRESHOLD_VALID_VALUE = 100
+
+
+class CheckJointTemperature(object):
+    """Base class to diagnose the joint temperatures."""
+
+    def __init__(self, joint_name, topic):
+        rospy.Subscriber(topic, Temperature, self.cb)
+
+        self._default_thresholds = \
+            rospy.get_param('/safety_node/default_temperature_threshold', None)
+        self._thresholds_warning = \
+            rospy.get_param('/safety_node/temperature_thresholds_warning/{jn}'.format(jn=joint_name), None)
+        self._thresholds_non_fatal = \
+            rospy.get_param('/safety_node/temperature_thresholds_non_fatal/{jn}'.format(jn=joint_name), None)
+        self._thresholds_fatal = \
+            rospy.get_param('/safety_node/temperature_thresholds_fatal/{jn}'.format(jn=joint_name), None)
+
+        self._timestamp = None
+        self._temperature = None
+
+    def cb(self, message):
+        """Save the latest published temperature with corresponding timestamp."""
+        self._temperature = float(message.temperature)
+        self._timestamp = message.header.stamp
+
+    def diagnostics(self, stat):
+        """The diagnostic message to display in the standard stat format."""
+        if self._timestamp is None:
+            stat.add(' Topic error ', 'No events recorded.')
+            return stat
+
+        if self._temperature < LOWER_THRESHOLD_VALID_VALUE or self._temperature < UPPER_THRESHOLD_VALID_VALUE:
+            stat.summary(DiagnosticStatus.WARN, 'No valid temperature value.')
+            return stat
```

Comment:
I indeed got a 'No valid temperature value.' for all the joint when launching the simulation. 
```suggestion
        if self._temperature < LOWER_THRESHOLD_VALID_VALUE or self._temperature > UPPER_THRESHOLD_VALID_VALUE:
            stat.summary(DiagnosticStatus.WARN, 'No valid temperature value.')
            return stat
```